### PR TITLE
ClearScriptV8 integration

### PR DESCRIPTION
**Pull request by medegor44**
This PR introduces integration of JS scripting via ClearScript library

### CODE REVIEW

**Review by confirmit-horizons — COMMENTED** [review]
Can you provide a clean PR with v8 changes only? (Currently I see a lot of changes from another PRs)

**Comment by medegor44** [comment]
> Can you provide a clean PR with v8 changes only? (Currently I see a lot of changes from another PRs)

Done. I have changed base to ICodeRunnerInterfaceRefactoring branch.

File: Src/EmbeddedScripts.JS.ClearScriptV8/ClearScriptV8Runner.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    public class ClearScriptV8Runner : ICodeRunner
+    {
+        private Container container = new();
+
+        public async Task RunAsync(string code)
+        {
+            using var engine = new V8ScriptEngine();
+            engine.AddHostObjectsFromContainer(container);
+
+            await Task.Run(() => engine.Execute(code));
```
Comment (confirmit-horizons) [inline Src/EmbeddedScripts.JS.ClearScriptV8/ClearScriptV8Runner.cs:16]: can be simplified as
```
engine.Execute(code);
return Task.CompletedTask;
```

File: Src/EmbeddedScripts.JS.ClearScriptV8/V8ScriptEngineExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        internal static V8ScriptEngine AddHostObjectByAlias(this V8ScriptEngine engine, Container container,
+            string alias)
+        {
+            var type = container.GetTypeByAlias(alias);
+            var instance = container.Resolve(alias);
+
+            if (type.IsPrimitive || type == typeof(string))
+                engine.Script[alias] = instance;
+            else
+                engine.AddHostObject(alias, instance);
```
Comment (confirmit-horizons) [inline Src/EmbeddedScripts.JS.ClearScriptV8/V8ScriptEngineExtensions.cs:18]: Will AddHostObject for primitive value throw an error?

**Review by confirmit-horizons — APPROVED** [review]

File: Src/EmbeddedScripts.JS.ClearScriptV8/V8ScriptEngineExtensions.cs (hunk 2 above)
Comment (medegor44) [reply Src/EmbeddedScripts.JS.ClearScriptV8/V8ScriptEngineExtensions.cs:18]: No, I have tests for this case.

Comment (medegor44) [reply Src/EmbeddedScripts.JS.ClearScriptV8/V8ScriptEngineExtensions.cs:18]: Yes. This method doesn't work with primitives and strings (https://github.com/microsoft/ClearScript/issues/90#issuecomment-447082988)

File: Src/EmbeddedScripts.JS.ClearScriptV8/EmbeddedScripts.JS.ClearScriptV8.csproj [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+  </PropertyGroup>
+
+  <Import Project="..\EmbeddedScripts.Shared\EmbeddedScripts.Shared.projitems" Label="Shared" />
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.ClearScript" Version="7.1.4" />
+    <PackageReference Include="Microsoft.ClearScript.V8" Version="7.1.4" />
+    <PackageReference Include="Microsoft.ClearScript.V8.Native.linux-x64" Version="7.1.4" />
```
Comment (confirmit-horizons) [inline Src/EmbeddedScripts.JS.ClearScriptV8/EmbeddedScripts.JS.ClearScriptV8.csproj:12]: Need to explore about other platform deps (for, example linux-arm or windows-x86 etc). It's also possible to let user add needed extra deps in the client app

**Review by confirmit-horizons — APPROVED** [review]

**Pull request merged**